firmware/esp32_ttgo/client/WebSocketClient.cpp

```diff
@@ -0,0 +1,91 @@
+#include "WebSocketClient.h"
+
+#include <unordered_map>
+
+void WebSocketClient::receive(String message)
+{
+    // TODO: msg id? defaulting to 0 for now
+    uint16_t msg_id = 0;
+
+    unsigned char buf[2 + message.length() + 2] = {'\0'};
+    memcpy(buf, &msg_id, 2);
```

Comment:
This is unnecessary. The msg_id does not need to be added by the WebSocket Client, it is already included in the message by the transmitting node. This code is analogous to the `sendToWS()` function from the original code, https://github.com/sudomesh/disaster-radio/blob/master/firmware/esp32_ttgo/main.ino#L88. Therefore it should only convert `message` from a `String` to an `unsigned char`.

Reply:
After testing, I see that it doesn't work with out that phony `msg_id` I think the reason is that in converting the received message (raw bytes) to a `String` type you lose the fact that the `msg_ID` is actually a two-byte binary value, instead it is represented by it's equivalent ASCII character.  Thus, by adding the `0` character to the beginning, the String is now the correct length to be accepted by the web app.

Maybe there is some sort of work around by parsing the String and converting it back in binary, but the correct way to fix this, which was already discussed in the thread, is to pass around structs of char arrays (or pointers to char arrays) as the messages between the server and clients.

I will leave the current bandage fix in for now. But it should be fixed ASAP as it could cause unexpected side effects.

Reply:
You'll see `LoRaClient` and `WebSocketClient` are the only clients that strip out message ID before calling `transmit()` and add in dummy message IDs in `receive()`, since those were the only existing clients I had to port.